src/pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -275,9 +275,18 @@ def _check_config_and_set_final_flags(
                     " you can use `Trainer(strategy='ddp_spawn', accelerator='tpu')` instead."
                 )
 
-        if accelerator is not None:
-            if accelerator in self._accelerator_types or accelerator == "auto" or isinstance(accelerator, Accelerator):
-                self._accelerator_flag = accelerator
+        if (
+            accelerator is not None
+            and accelerator not in self._accelerator_types
+            and accelerator != "auto"
+            and not isinstance(accelerator, Accelerator)
+        ):
+            raise ValueError(
+                f"You selected an invalid accelerator name: `accelerator='{accelerator}'`."
```

Comment:
```suggestion
                f"You selected an invalid accelerator name: `accelerator={accelerator!r}`."
```
can be any primitive type.